### PR TITLE
Add auth middleware functions

### LINKED ISSUE #10: [ MISC ] - Auth Layer Middleware Function
We need a common middleware function on our backend that acts as an auth layer. This function should check to make sure that a valid user is accessing the route and should update the req body to add the type of user accessing the route. This way we can ensure that all of our functions are protected and that no one can steal data.

References:

[See Auth Layer From ALUM](https://github.com/TritonSE/ALUM-Mobile-Application/blob/main/backend/src/middleware/auth.ts)

[Auth Layer PR From ALUM](https://github.com/TritonSE/ALUM-Mobile-Application/pull/73)

[Firebase Documentation](https://firebase.google.com/docs/auth/admin/verify-id-tokens)

### PR DESCRIPTION
**Pull request by vs2961**
## Tracking Info

Resolves #10 

## Changes

Added the middleware function for user token authentication, as well as any supporting functions and imports.

## Testing

Not sure how to test it without any routes at the moment, but I'll make a follow-up once some are implemented.

### CODE REVIEW

File: backend/.eslintrc.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
     // Uncomment this if you have verbatimModuleSyntax enabled and importing
     // .ts files with a .js extension gives lint errors.
-    // "import/no-unresolved": "off",
+    "import/no-unresolved": "off",
```
Comment (vs2961) [inline backend/.eslintrc.json:34]: I couldn't get firebase to import properly, and people online say that we either need a custom import solution to fix this. This was the easiest fix I found, but it's a temporary patch.

Comment (adhi0331) [reply backend/.eslintrc.json:34]: This should be fine. If you look in the comments above we should be able to use this eslint feature

**Review by adhi0331 — APPROVED** [review]
LGTM!!! 🚀

**Review by aammya8 — APPROVED** [review]
Looks good!

**Pull request merged**